### ISSUE TITLE
Artificial Intelligence

### ISSUE THREAD

**Issue by mentifex** [BuildingXwithJS/proposals#22]

**Comment by yamalight** [comment]
AI is an interesting area, but the area itself it a bit too wide for one livestream :)
Do you have any specific things in mind?

**Comment by mentifex** [comment]
Yes, I had written a whole page, but by some glitch it disappeared. 

The "Example proposals" at https://github.com/BuildingXwithJS/proposals include "Review this bit of new cool tech" which covers the idea of looking into the area of artificial intelligence in JavaScript. 

My JavaScript AI at http://ai.neocities.org/MsIeAi.html raises several issues, each one potentially worthy of a video -- such as https://www.youtube.com/watch?v=QuBEgINpGdo of an obsolete JSAI -- or a livestream. 1) Is the JavaScript [**AI Mind**](http://ai.neocities.org/index.html) true AI, that is, does it really think when it does automated reasoning by logical inference? You can type in, "mary is a woman" and the JSAI will ask you, "Does Mary have a child?" by inferring the possibility. 

2) Do the security provisions of client-side JavaScript address some of the issues of the [**Control Problem**](http://www.reddit.com/r/ControlProblem) for AI?

3) Since the JavaScript [**AI Mind**](http://ai.neocities.org/MsIeAi.html) is almost a compleat translation of the [**MindForth**](http://dl.acm.org/citation.cfm?doid=307824.307853) and [**ghost**](http://wiki.opencog.org/w/Ghost).pl AI programs, is AI in JavaScript a good resource for teaching classes about artificial intelligence and about JavaScript? 

4) In view of the above, is JavaScript AI a good tool for training in the hopefully imminent field of [**AI Mind Maintainer**](http://ai.neocities.org/maintainer.html)?

**Comment by yamalight** [comment]
That's is a lot of links to digest! I'll need some time for it 😄 
Is the code for your implementation available somewhere?

**Comment by mentifex** [comment]
http://ai.neocities.org/MsIeAi.html -- is the implementation code

MsIeAi is M(icro)s(oft) I(nternet) E(xplorer) A(rtificial) I(ntelligence).

**Comment by yamalight** [comment]
Ah, okay, everything is in the sources there. Got it!

**Comment by yamalight** [comment]
Alright, so I have read through most of the things you've referenced.
I'm going to accept this proposal, but I think I'll need a lot more time to prepare for stream/video on it.
I am far from being an expert on AI topic, but I do have colleagues I can discuss this with. We're going to see where it all goes 😄

**Comment by FDiskas** [comment]
https://experiments.withgoogle.com/ai

**Comment by ArcticSpaceFox** [comment]
AI is very cool and promising! Maybe do some fraud detection with it.

**Comment by FDiskas** [comment]
I was trying to do some testing with text and ai results were bad. Best results are with images. So my suggestion would be to try to do some app who would inform you with an alert when you touching your face using camera. Do to Corona virus prevention.
I know that there's is some similar web aps so lets do it with electron or other desctop app.

**Comment by ArcticSpaceFox** [comment]
Cool idea :+1:
